### DISCUSSION TITLE
Pluck could have been retrieved as query but how?

### DISCUSSION THREAD

**Discussion by akulmehta** [larastan/larastan#1326] [Q&A] [answered]
I have a statement which gets a list of users as an array

```php
User::select('name')->get()->pluck('name')->toArray()
```

This results in:
```php
 [
     "Test",
     "Reviewer",
]
```

Larastan gives the following error:

```
Called 'pluck' on Laravel collection, but could have been retrieved as a query.
```

If I don't use the pluck using `User::select('name')->get()->toArray()` I get an array of associative arrays like so:
```php
[
     [
       "name" => "Test",
     ],
     [
       "name" => "Reviewer",
     ],
]
```
I checked the Rules at https://github.com/nunomaduro/larastan/blob/master/docs/rules.md#examples-1 but that only shows an example where existence is checked. 

So what query do I use to get the array like I desired above?

**Comment by szepeviktor** [comment, marked as answer]
How about `User::select('name')->pluck('name')->get()->toArray()`?

**Reply by akulmehta** [reply]
Gives TypeError

```
TypeError: Too few arguments to function Illuminate\Support\Collection::get(), 0 passed in .... at least 1 expected
```

**Reply by szepeviktor** [reply]
I'm not a developer! Do not listen to me.

`User::select('name')->pluck('name')->toArray()` works for me

**Reply by akulmehta** [reply]
That works... thanks.

**Reply by eithed** [reply]
Today I found out that it can be shortened to `User::pluck('name')->toArray()`
